libs/ledger-live-common/src/families/cosmos/js-prepareTransaction.ts

```diff
@@ -42,10 +48,39 @@ const getEstimatedFees = async (
 ): Promise<any> => {
   const cosmosCurrency = cryptoFactory(account.currency.id);
   let estimatedGas = new BigNumber(cosmosCurrency.defaultGas);
-  if (cosmosCurrency.gas[transaction.mode]) {
-    estimatedGas = new BigNumber(cosmosCurrency.gas[transaction.mode]);
+
+  const cosmosAPI = new CosmosAPI(account.currency.id);
+  const unsignedPayload = await buildTransactionWithUnsignedPayload(
+    account,
+    transaction
+  );
+
+  if (unsignedPayload && unsignedPayload.length > 0) {
+    const pubkey = {
+      typeUrl: "/cosmos.crypto.secp256k1.PubKey",
+      value: new Uint8Array([
+        ...new Uint8Array([10, 33]),
+        ...new Uint8Array(Buffer.from(account.seedIdentifier, "hex")),
+      ]),
```

Comment:
This isn't really self-explanatory, would be worth a comment

Reply:
Done